src/main/java/gov/cabinetoffice/gap/adminbackend/services/GrantMandatoryQuestionService.java

```diff
@@ -0,0 +1,163 @@
+            return StringUtils.defaultString(addressLine2);
+        }
+
+        return String.join(", ", StringUtils.defaultString(addressLine1), StringUtils.defaultString(addressLine2));
+    }
+
+    /**
+     * The ordering of the data added here is strongly tied to SPOTLIGHT_HEADERS. If new
+     * headers are added or the ordering is changed in SPOTLIGHT_HEADERS, this will need
+     * manually reflected here.
+     */
+    public List<String> buildSingleSpotlightRow(GrantMandatoryQuestions grantMandatoryQuestions) {
+        try {
+
+            final Integer schemeId = grantMandatoryQuestions.getSchemeEntity().getId();
+
+            final String gapId = grantMandatoryQuestions.getGapId();
+            final String organisationName = grantMandatoryQuestions.getName();
+            final String addressStreet = combineAddressLines(grantMandatoryQuestions.getAddressLine1(),
+                    grantMandatoryQuestions.getAddressLine2());
+            final String addressTown = grantMandatoryQuestions.getCity();
+            final String addressCounty = grantMandatoryQuestions.getCounty();
+            final String postcode = grantMandatoryQuestions.getPostcode();
+            final String charityNumber = grantMandatoryQuestions.getCharityCommissionNumber();
+            final String companyNumber = grantMandatoryQuestions.getCompaniesHouseNumber();
+            final String applicationAmount = grantMandatoryQuestions.getFundingAmount() == null ? null
+                    : grantMandatoryQuestions.getFundingAmount().toString();
+
+            List<String> row = new ArrayList<>();
+            row.add(mandatoryValue(schemeId, "gap id", gapId));
+            row.add(mandatoryValue(schemeId, "organisation name", organisationName));
+            row.add(addressStreet);
+            row.add(addressTown);
+            row.add(addressCounty);
+            row.add(mandatoryValue(schemeId, "postcode", postcode));
+            row.add(mandatoryValue(schemeId, "application amount", applicationAmount));
+            row.add(charityNumber);
+            row.add(companyNumber);
+            row.add(""); // similarities data - should always be blank
+
+            return row;
+        }
+        catch (NullPointerException e) {
+            throw new SpotlightExportException("Unable to find mandatory question data: " + e.getMessage());
+        }
+    }
+
+    public String generateExportFileName(Integer schemeId) {
+        SchemeDTO schemeDTO = schemeService.getSchemeBySchemeId(schemeId);
+        String ggisReference = schemeDTO.getGgisReference();
+        String schemeName = schemeDTO.getName().replace(" ", "_").replaceAll("[^A-Za-z0-9_]", "");
+        String dateString = new SimpleDateFormat("yyyy-MM-dd").format(Calendar.getInstance().getTime());
+
+        return dateString + "_" + ggisReference + "_" + schemeName + ".xlsx";
+    }
+
+    public boolean doesSchemeHaveCompletedMandatoryQuestions(Integer schemeId) {
```

Comment:
Same again here with the naming conventions for methods that return boolean values